Automattic/jetpack
# [RNMobile] Add Playback Bar Color Settings panel to VideoPress Block Settings

Part of an ongoing project to add the VideoPress block to the native apps.

## Proposed changes:

* A native version of the `ColorPanel` component has been introduced and added to the VideoPress block's settings. The main code changes in the component's file can be found in https://github.com/Automattic/jetpack/pull/29567/commits/39c1da60cbeaab7ae2676807db116cbe889fba75.

### Other information:


## Jetpack product discussion

p9ugOq-3mV-p2

## Does this pull request change what data or activity we track or use?

No, it does not.

## Testing instructions:

* Go through the steps to add the VideoPress block on the web, optionally toggle the playback bar colors.
* Open up the app and navigate to the block that was added via the web.
* Tap the settings toggle then tap `Dynamic Color` to open the new settings panel.
* If you'd changed the video's playback bar colors on the web, verify that these are accurately reflected in the panel on native.
* Experiment with the new color settings by either toggling `Dynamic Color` on or selecting colors manually when that toggle is off. 
* Ensure the updated colors are accurately reflected in the video's HTML markup (by switching to the editor's HTML mode). Optionally, also open the block back up on web to verify that the colors are updated in the settings there.

### Screenshots

| Subsheet (toggle on)  | Subsheet (toggle off) |
| ------------- | ------------- |
| <img src="https://user-images.githubusercontent.com/2998162/226430326-19d7c3e0-e29f-46ca-bb28-969bfd91bd21.png" width="100%"> | <img src="https://user-images.githubusercontent.com/2998162/226430198-a0b3c3d3-565b-49f5-a1aa-e83cfb36aa41.png" width="100%"> |

| Color panel  | Main sheet |
| ------------- | ------------- |
| <img src="https://user-images.githubusercontent.com/2998162/226430457-3e69ce7d-0c45-484e-9f4e-b558a9541e53.png" width="100%"> | <img src="https://user-images.githubusercontent.com/2998162/226430829-bdeb89aa-3196-4a14-a8c0-4f4b9074ccc3.png" width="100%"> |

### Screen recording

https://user-images.githubusercontent.com/2998162/226431528-6a329595-1c46-44e8-8873-68dfb1033ed7.mov
